Added support of multiple special groups in regexp

drey opened the pull request:
Hi Alexandr, you made a really useful plugin. I have tried to do it a little bit better.

Now it's possible to use special group (!...) several times. For example, you can search by regular expression like that: "(!ip\sdhcp\spool.+?)\n(?:.+?\n)+?(!\sdns-server.+$)", to get result showed below:

![screen shot 2016-09-10 at 15 03 53](https://cloud.githubusercontent.com/assets/157472/18410377/6ebd6ec0-7772-11e6-8b69-7f84ea759afe.png)

Thanks!

Commit:
Added support of multiple special groups in regexp

Now it's possible to use special group (!...) several times.
Removed same code duplication by adding a base class.

mvoidex commented:
Nice feature, thanks!
I'll review it (all tabs were replaced with spaces, so diff is useless now), update readme and merge.

Commit:
Remove selection by special regexp group notation

It's possible selectiong groups by standart regexp group notation. To avoid regexp groups selection you can use embedded rexexp synax "(?:...)".

drey commented:
I implemented a feature to select by regexp group without special group notation. I prefer to use classical regexp syntax to work with groups. And of course you can exclude a group from selection by embedded regexp method `(?:..)`. You can merge it if you like.

ps You can see diff between our latest commits by this [link](https://github.com/mvoidex/SublimeSelectByRegex/compare/master...Drey:feature-multiple-groups).

mvoidex commented:
Merged, thanks for PR!

mvoidex commented on selectbyregex.py:
```diff
@@ -2,53 +2,44 @@
+			self.rx = re.compile(unwrap_sel(rx, self.view.substr(r)), re.MULTILINE)
+			m = self.rx.search(self.text, r.a)
+			if m and r.a == r.b or m.end() <= r.b:  # Restrict to selection if it's not empty
+				if len(m.groups()) > 1:
```
Maybe `len(m.groups()) > 0`?

drey replied:
Yes, you are right! And if to be honest may be `len(m.groups())` will be clearer.

Thanks!
